text/0000-otlp-trace-data-format.md

```diff
@@ -0,0 +1,376 @@
+# OTLP Trace Data Format
+
+_Author: Tigran Najaryan, Omnition Inc._
+
+**Status:** `proposed`
+
+OTLP Trace Data Format specification describes the structure of the trace data that is transported by OpenTelemetry Protocol (RFCNNN).
+
+## Motivation
+
+This document is a continuation of OpenTelemetry Protocol RFCNNN and is necessary part of OTLP specification.
+
+## Explanation
+
+OTLP Trace Data Format is primarily inherited from OpenCensus protocol. Several changes are introduced with the goal of more efficient serialization. Notable differences from OpenCensus protocol are:
+
+1. Removed `Node` as a concept.
+2. Extended `Resource` to better describe the source of the telemetry data.
+3. Replaced attribute maps by lists of key/value pairs.
+4. Eliminated unnecessary additional nesting in various values.
+
+Changes 1-2 are conceptual, changes 3-4 improve performance.
+
+## Internal details
+
+This section specifies data format in Protocol Buffers.
+
+### Resource
+
+```
+// Resource information. This describes the source of telemetry data.
+message Resource {
+  // Identifier that uniquely identifies a process within a VM/container.
+  Process process = 1;
+
+  // Name of the service.
+  string service_name = 2;
```

Comment:
Why is "service" important that gets a separate treatment?

Reply:
It is the human-readable name of the service or application that is emitting the telemetry. I think it is important for visualization purposes. In OpenCensus this was in ServiceInfo message (which didn't have any other fields) so I moved it here.

Reply:
As I mentioned previously `service_name` should be removed and we should have a way to include `service.name` and `service.task_id` in the attributes. 

No reason why `service_name` is more important than any other information in the `attributes`, please try to not be primarily focus on Jaeger (which may or may not be the only backend/protocol that vendor use).

Reply:
`service_name` removed now.